### DISCUSSION TITLE
Confusing naming of struct fields or methods   RVV-B0001

### DISCUSSION THREAD

**Discussion by xiaods** [k3s-io/k3s#4548] [General]
use deepsource.io, it reports some potential problems below:

pkg/etcd/etcd.g
<img width="819" alt="截屏2021-11-20 下午11 54 22" src="https://user-images.githubusercontent.com/37678/142732755-ccc638a1-5f86-4fdd-9612-82340cc3ec42.png">

<img width="828" alt="截屏2021-11-20 下午11 55 05" src="https://user-images.githubusercontent.com/37678/142732787-118cdabc-3000-4d13-9e0b-be5afbeb17b5.png">

<img width="819" alt="截屏2021-11-20 下午11 55 21" src="https://user-images.githubusercontent.com/37678/142732798-8957ce21-ee87-4986-a91b-c3b8b09bbfa2.png">
<img width="838" alt="截屏2021-11-20 下午11 55 42" src="https://user-images.githubusercontent.com/37678/142732809-63027aeb-352c-449c-af27-00ff0a716c10.png">

My question is whether I can improve these codes?

**Comment by brandond** [comment]
Most of these are due to exposed methods wrapping more complicated internal implementation methods.

You're welcome to open a PR if you think there are improvements to be made.

**Reply by xiaods** [reply]
My side project is based on k3s upsteam. I will create a pr for improve k3s codebase asap.

**Comment by dweomer** [comment]
deepsource.io doesn't seem aware of how folks write Go code.
